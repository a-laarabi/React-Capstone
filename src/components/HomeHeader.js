import React from 'react';
import videoWatch from '../Watch_Dogs2-Trailer.mp4';
import '../styles/HomeHeader.css';

function HomeHeader() {
  return (
    <header className="homeHeader">
      <h1 className="Logo">Your New Game</h1>
      <ul className="headerInfo">
        <li className="headerInfo_Title">Watch Dogs</li>
        <li>Action-aventure</li>
        <li>2014-05-27</li>
      </ul>
      <video src={videoWatch} muted loop autoPlay />
    </header>
  );
}

export default HomeHeader;
