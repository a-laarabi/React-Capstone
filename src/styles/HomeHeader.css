@media (max-width: 768px) {
  .homeHeader {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 40%;
    background: #1d1d1d;
    mix-blend-mode: overlay;
  }

  .Logo {
    margin-top: 10px;
    text-align: center;
    color: #fff;
    font-family: 'Ms Madi', cursive;
    position: relative;
  }

  .homeHeader video {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    object-fit: cover;
    z-index: -1;
    opacity: 0.6;
  }

  .headerInfo {
    color: #fff;
    margin-top: 30%;
    margin-left: 20px;
    padding-bottom: 30px;
  }

  .headerInfo li {
    font-size: 12px;
    color: rgb(193, 194, 196);
  }

  .headerInfo li:first-child {
    font-weight: bold;
    font-size: 30px;
    color: #fff;
  }
}