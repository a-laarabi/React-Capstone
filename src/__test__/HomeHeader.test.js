import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/store';
import HomeHeader from '../components/HomeHeader';

describe('Test the components', () => {
  it('Check if the HomeHeader.js file\'s section are available', () => {
    const website = renderer.create(
      <Provider store={store}>
        <HomeHeader />
      </Provider>,
    );
    expect(website).toMatchSnapshot();
  });
});
