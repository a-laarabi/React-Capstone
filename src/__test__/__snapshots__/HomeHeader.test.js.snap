// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the components Check if the HomeHeader.js file's section are available 1`] = `
<header
  className="homeHeader"
>
  <h1
    className="Logo"
  >
    Your New Game
  </h1>
  <ul
    className="headerInfo"
  >
    <li
      className="headerInfo_Title"
    >
      Watch Dogs
    </li>
    <li>
      Action-aventure
    </li>
    <li>
      2014-05-27
    </li>
  </ul>
  <video
    autoPlay={true}
    loop={true}
    muted={true}
    src="Watch_Dogs2-Trailer.mp4"
  />
</header>
`;
